### DISCUSSION TITLE
Vacuum JSON report or import into nodejs project

### DISCUSSION THREAD

**Discussion by sunday-oyeniyi6** [daveshanley/vacuum#516] [Q&A] [answered]
Hello community,
thank you for this amazing tool for OPENAPI specification.

it's possible to generate vacuum report as JSON object or it's possible to be imported into nodejs project ?

Regards,

**Comment by LasneF** [comment, marked as answer]
have you check the documentation there ? 
https://quobix.com/vacuum/commands/report/ 

as you can see you can generate a report , either following JUnit format (XML) , then there are plenty of tooling allowing to display Junit  in a nice way   , or  leveraging JSON 

this by applying the simple command line ``vacuum report .\component-descriptor.yml toto.json`` 

This would generate stuff like this 
```

 {
                "message": "schema is missing `examples` or `example`",
                "range": {
                    "start": {
                        "line": 169,
                        "character": 19
                    },
                    "end": {
                        "line": 169,
                        "character": 21
                    }
                },
                "path": "$.paths['/trains'].post.responses['201'].content['application/json'].schema.properties['id']",
                "ruleId": "oas3-missing-example",
                "ruleSeverity": "warn"
            },

```

notice that here you have several option to leverage the output either as a file , either to sdtout that may help you 

take care of the report naming convention in case you use file

**Reply by sunday-oyeniyi6** [reply]
Hello @LasneF ,
Thank you for your answer.

**Comment by achsoftathome** [comment]
Hi, I am confused

1 - You have a command to lint the yaml files:
vacuum lint --no-banner -d -q --no-clip --ext-refs -c -d api/**/*.yaml 
so the output is the console, that is useless for auto interpretation.

2 - You have another command to generate report that don't accept wild card search:
vacuum report .\component-descriptor.yml toto.json

This call always generates en empty report even if lint found error:
{
    "generated": "2025-03-03T12:22:33.214242485+01:00",
    "specInfo": null,
    "statistics": null,
    "resultSet": {
        "warningCount": 0,
        "errorCount": 0,
        "infoCount": 0
    }
}

3 - How mix both with consistent result?

Is there any processing pipe: lint (*.yaml) > report > renderer (json)
